NodeJS/src/AgentClient.ts

```diff
@@ -0,0 +1,413 @@
+        }
+
+        let subAccountId = await this.getSubAccountId();
+        let station = {};
+        try {
+            station = await this.getStationOnly(subAccountId, agentUsername);
+            if (!station) {
+                console.log('station associated with ' + agentUsername + ' not found');
+                station = {}
+            }
+        } catch (e) {
+            if (e.response.status === 404) {
+                console.log('station associated with ' + agentUsername + ' not found')
+            } else {
+                throw e
+            }
+        }
+
+        return {'agent':agent, 'station': station}
+    }
+
+    async getSubAccountId() {
+        return this.session.getSubAccount()
+            .then((response: { id: any }) => {
+                return response.id
+            })
+    }
+
+    async getAgentByUsername(agent_username: string) {
+        return this.session.get('/spokenAbc/agents/username/' + agent_username)
+            .then((response: { data: any }) => {
+                // console.log(response.data)
+                return response.data
+            })
+    }
+
+    async getAgentByLoginId(loginId: string) {
+        return this.session.get('/spokenAbc/agents/loginId/' + loginId)
+            .then((response: { data: any }) => {
+                // console.log(response.data)
+                return response.data
+            })
+    }
+
+    async waitForAgentCreation(loginId: string) {
+        return new Promise((resolve, reject) => {
+            process.stdout.write("Creating agent.");
+            let counter = 0;
+            const intervalId = setInterval(async () => {
+                try {
+                    await this.getAgentByLoginId(loginId);
+                    console.log('agent created');
+                    clearInterval(intervalId);
+                    resolve()
+                } catch (e) {
+                    if (e.response.status === 404) {
+                        process.stdout.write('.');
+                        counter++;
+                        if (counter > MAX_RETRY) {
+                            console.log();
+                            clearInterval(intervalId);
+                            reject(Error('agent creation failed'))
+                        }
+                    } else {
+                        console.log(e);
+                        throw e
+                    }
+                }
+            }, INTERVAL_IN_MILLIS);
+            return intervalId
+        })
+    }
+
+    async waitForAgentDeletion(agent_username: string) {
+        return new Promise((resolve, reject) => {
+            process.stdout.write("Deleting agent.");
+            let counter = 0;
+            const intervalId = setInterval(async () => {
+                try {
+                    await this.getAgentByUsername(agent_username);
+                    process.stdout.write('.');
+                    counter++;
+                    if (counter > MAX_RETRY) {
```

Comment:
retry logic here?  What if a non 404 exception is thrown?

Reply:
For now we are not doing any retry. 
For non 404 we are throwing the exception that stops the execution.